src/legacy/core_plugins/kibana/public/management/sections/objects/lib/__jest__/process_import_response.test.js

```diff
@@ -0,0 +1,188 @@
+ * under the License.
+ */
+
+import { processImportResponse } from '../process_import_response';
+
+describe('processImportResponse()', () => {
+  test('works when no errors exist in the response', () => {
+    const response = {
+      success: true,
+      successCount: 0,
+    };
+    const result = processImportResponse(response);
+    expect(result).toMatchInlineSnapshot(`
+Object {
+  "conflictedSavedObjectsLinkedToSavedSearches": undefined,
+  "conflictedSearchDocs": undefined,
+  "failedImports": Array [],
+  "importCount": 0,
+  "status": "success",
+  "unmatchedReferences": Array [],
+}
+`);
+  });
+
+  test('conflict errors get added to failedImports', () => {
+    const response = {
+      success: false,
+      successCount: 0,
+      errors: [
+        {
+          obj: {
+            type: 'a',
+            id: '1',
+          },
+          error: {
+            type: 'conflict',
+          },
+        },
+      ],
+    };
+    const result = processImportResponse(response);
+    expect(result).toMatchInlineSnapshot(`
+Object {
+  "conflictedSavedObjectsLinkedToSavedSearches": undefined,
+  "conflictedSearchDocs": undefined,
+  "failedImports": Array [
+    Object {
+      "error": Object {
+        "type": "conflict",
+      },
+      "obj": Object {
+        "obj": Object {
+          "id": "1",
+          "type": "a",
+        },
+      },
+    },
+  ],
+  "importCount": 0,
+  "status": "idle",
+  "unmatchedReferences": Array [],
+}
+`);
+  });
+
+  test('unknown errors get added to failedImports', () => {
+    const response = {
+      success: false,
+      successCount: 0,
+      errors: [
+        {
+          obj: {
+            type: 'a',
+            id: '1',
+          },
+          error: {
+            type: 'unknown',
+          },
+        },
+      ],
+    };
+    const result = processImportResponse(response);
+    expect(result).toMatchInlineSnapshot(`
+Object {
+  "conflictedSavedObjectsLinkedToSavedSearches": undefined,
+  "conflictedSearchDocs": undefined,
+  "failedImports": Array [
+    Object {
+      "error": Object {
+        "type": "unknown",
+      },
+      "obj": Object {
+        "obj": Object {
+          "id": "1",
+          "type": "a",
+        },
+      },
+    },
+  ],
+  "importCount": 0,
+  "status": "success",
+  "unmatchedReferences": Array [],
+}
+`);
+  });
+
+  test('missing references get added to failedImports', () => {
+    const response = {
+      success: false,
+      successCount: 0,
+      errors: [
+        {
+          obj: {
+            type: 'a',
+            id: '1',
+          },
+          error: {
+            type: 'missing_references',
+            references: [
+              {
+                type: 'index-pattern',
+                id: '2',
+              },
+            ],
+          },
+        },
+      ],
+    };
+    const result = processImportResponse(response);
+    expect(result).toMatchInlineSnapshot(`
```

Comment:
Looks like this only needs to assert that `result.failedImports` (maybe `unmatchedReferences` too) is correct? We should limit the scope of the snapshots so there aren't unrelated failures in the future.